internal/services/appservice/windows_function_app_resource.go

```diff
@@ -546,6 +573,43 @@ func (r WindowsFunctionAppResource) Read() sdk.ResourceFunc {
 				return fmt.Errorf("reading App Settings for Windows %s: %+v", id, err)
 			}
 
+			var storageAccountName, storageFileShare string
+			if sendContentSettings || helpers.PlanIsConsumption(planSku) {
+				if appSettingsResp.Properties != nil {
+					if appSettingsResp.Properties["WEBSITE_CONTENTAZUREFILECONNECTIONSTRING"] != nil {
+						saConnectionString := strings.Split(*appSettingsResp.Properties["WEBSITE_CONTENTAZUREFILECONNECTIONSTRING"], ";")
+						for _, part := range saConnectionString {
+							if strings.Contains(part, "AccountName") {
+								storageAccountName = strings.TrimPrefix(part, "AccountName=")
+								break
+							}
+						}
+					}
+					if appSettingsResp.Properties["WEBSITE_CONTENTSHARE"] != nil {
+						storageFileShare = *appSettingsResp.Properties["WEBSITE_CONTENTSHARE"]
+					}
+					saClient := metadata.Client.Storage
+					account, err := saClient.FindAccount(ctx, storageAccountName)
+					if err != nil {
+						return fmt.Errorf("retrieving Account %q for Share %q: %s", storageAccountName, storageFileShare, err)
+					}
+					if account == nil {
+						return fmt.Errorf("unable to locate Storage Account %s", storageAccountName)
+					}
+					saFileShareClient, err := metadata.Client.Storage.FileSharesClient(ctx, *account)
+					if err != nil {
+						return fmt.Errorf("building File Share Client: %s", err)
+					}
+					exists, err := saFileShareClient.Exists(ctx, account.ResourceGroup, storageAccountName, storageFileShare)
+					if err != nil {
+						return fmt.Errorf("checking for existence of Storage Share %q (Account %q) : %+v", storageFileShare, storageAccountName, err)
+					}
+					if exists == nil {
+						return fmt.Errorf("the storage file share %s for function app %s does not exists, please specify an existing one", storageFileShare, id.ID())
+					}
+				}
+			}
+
```

Comment:
As above, this shouldn't be in the Read.

Reply:
Thanks for the review, this is to address the timeout issue. if the user doesn't specify a predefined file share and also set the WEBSITE_CONTENTOVERVNET in the app_setting, the provider will timeout by trying to access a non-existence file share. I can add the 4.0 condition to this check block.

Reply:
agreed - this shouldn't be in the Read function - we don't retrieve the file share in the Read, so this is unnecessary

Reply:
Thanks @tombuildsstuff for the comments. If we don't check the storage file share, shall we just let the read function timeout?

Reply:
@xiaxyi why will the read function timeout? it's calling the ARM API and interacting with the Storage API at all?

Reply:
The function app is trying to connect to the file share specified in the `[WEBSITE_CONTENTSHARE](https://learn.microsoft.com/en-us/azure/azure-functions/functions-app-settings)`, Generally, the file share is created by function app automatically, however, the it cannot be created if user specify the key `WEBSITE_CONTENTOVERVNET` in `app_setting, it requires the user to create a valid file share in advance and put the value in `WEBSITE_CONTENTSHARE`. We didn't add such notes to user in the current provider, so the function(Read function) times out.

If we don't check the file share in read function, then what's left is to add notes in docs and let the read function timeout.